indico/modules/events/management/controllers/protection.py

```diff
@@ -71,34 +101,85 @@ class RHEventProtection(RHManageEventBase):
             flash(_('Protection settings have been updated'), 'success')
             return redirect(url_for('.protection', self.event))
         return WPEventProtection.render_template('event_protection.html', self.event, 'protection', form=form)
 
     def _get_defaults(self):
-        acl = {p.principal for p in self.event.acl_entries if p.read_access}
-        submitters = {p.principal
-                      for p in self.event.acl_entries
-                      if p.has_management_permission('submit', explicit=True)}
-        managers = {p.principal for p in self.event.acl_entries if p.full_access}
         registration_managers = {p.principal for p in self.event.acl_entries
                                  if p.has_management_permission('registration', explicit=True)}
         event_session_settings = session_settings.get_all(self.event)
         coordinator_privs = {name: event_session_settings[val] for name, val in COORDINATOR_PRIV_SETTINGS.iteritems()
                              if event_session_settings.get(val)}
-        return dict({'protection_mode': self.event.protection_mode, 'acl': acl, 'managers': managers,
-                     'registration_managers': registration_managers, 'submitters': submitters,
+        permissions = [[serialize_principal(p.principal), list(self._get_principal_permissions(p))]
+                       for p in self.event.acl_entries if self._get_principal_permissions(p)]
+
+        return dict({'protection_mode': self.event.protection_mode, 'registration_managers': registration_managers,
                      'access_key': self.event.access_key, 'visibility': self.event.visibility,
-                     'own_no_access_contact': self.event.own_no_access_contact}, **coordinator_privs)
+                     'own_no_access_contact': self.event.own_no_access_contact, 'permissions': permissions},
+                    **coordinator_privs)
 
     def _update_session_coordinator_privs(self, form):
         data = {field: getattr(form, field).data for field in form.priv_fields}
         update_session_coordinator_privs(self.event, data)
+
+    def _get_principal_permissions(self, principal):
+        """Retrieve a set containing the valid permissions of a principal."""
+        permissions = set()
+        if principal.full_access:
+            permissions.add(FULL_ACCESS_PERMISSION)
+        if principal.read_access:
+            permissions.add(READ_ACCESS_PERMISSION)
+        available_permissions = get_permissions_info()[0]
+        return permissions | (set(principal.permissions) & set(available_permissions))
+
+    def _update_permissions(self, current, new):
+        """Handle the updates of permissions and creations/deletions of acl principals.
+
+        :param current: A dict mapping principals and a set with its current permissions
+        :param new: A dict mapping principals and a set with its new permissions
+        """
+        def _get_splitted_permissions(_permissions):
```

Comment:
does this need to be nested inside the method? also, the `_permissions` argument name is a bit weird IMO